### DISCUSSION TITLE
Wifi Network not listed

### DISCUSSION THREAD

**Discussion by Kalisov** [KlipperScreen/KlipperScreen#1279] [Q&A] [answered]
Hi everyone! I have a problem with Network Screen on Klipper Screen. When I open it i can only see the current connection and can't see other available wifi networks. I was trying this: https://klipperscreen.readthedocs.io/en/latest/Troubleshooting/Network/ but i have different error in logs. I run Klipper on small Thin Client PC with Debian 12 installed. Network manager is installed but it looks like Klipper Screen can't find network interface for some reason. Does someone know what could be the problem?

Here's the fragment of KlipperScreen.log:

`2024-02-18 19:48:27,699 [screen.py:_load_panel()] - Loading panel: network
2024-02-18 19:48:27,739 [network.py:__init__()] - Nie znaleziono żadnego bezprzewodowego interfejsu (ang: No wireless interface found)
2024-02-18 19:48:27,767 [screen.py:attach_panel()] - Current panel hierarchy: main_menu > more > network`

**Comment by alfrix** [comment]
what is the output of `nmcli device status`?

**Reply by Kalisov** [reply]
```
pi@ender3v2:~$ nmcli device status
DEVICE         TYPE      STATE                   CONNECTION
mlan0          wifi      połączono               videolan_HALNy-2.4G-8dd4bf
lo             loopback  connected (externally)  lo
p2p-dev-mlan0  wifi-p2p  rozłączono              --
enp1s0         ethernet  niedostępne             --
```

połączono means: connected
rozłączono means: disconnected
niedostępne means: unavailable

I can connect to different wifi networks using for example `nmtui`.

**Reply by alfrix** [reply, marked as answer]
that is the issue, the current network backend looks for devices starting with `wl` as it's the standard naming, i will not code any fixes to the current backend, since i'm developing the new backend that can filter by type, to fix your issue rename the iface to wlan0

**Reply by Kalisov** [reply]
Thank You so much! It's now working perfectly! 
For future reference i used this: 
https://forty.sh/posts/2021-01-15-renaming-network-interfaces/ 
to change the interface name from mlan0 to wlan0

**Reply by erwann17** [reply]
Hello I have the same PB and I can't solve it, I'm currently on Debian Sonic Pad
![image](https://github.com/KlipperScreen/KlipperScreen/assets/146946050/d9371a24-7a68-4bdd-a6fa-8df3e42ac5bf)

**Reply by erwann17** [reply]
please help me

**Reply by martin-badin** [reply]
@alfrix this issue still persist. Do you have a plan to fix that? I found some closed issues but it doesn't help me. It seams connection works through `nmtui` but not through KlipperScreen. And yes, I am using the latest version of KlipperScreen v0.4.6-12-gb3115f9b

**Comment by alfrix** [comment]
the original issue was solved by a new backend based on network-manager, new discussions can be created for new issues
